File: EMS邮惠中心.js
/*
cron "0 9 * * *" emsyhzx.js
export EMSYHZX_OPENID='第一个openId&第二个openId'
微信小程序:EMS中国邮政速递物流
*/

const $ = new Env('EMS邮惠中心');
const notify = $.isNode() ? require('../sendNotify') : '';
const EMSYHZX_OPENID = ($.isNode() ? process.env.EMSYHZX_OPENID : $.getdata("EMSYHZX_OPENID")) || '';

let token = '', userId = '';
let notice = '';
const maxRetries = 3; // 网络请求重试次数

// 完整 Env 类，确保兼容性
function Env(name, opts) {
  class Http {
    constructor(env) {
      this.env = env
    }

    send(opts, method = 'GET') {
      opts = typeof opts === 'string' ? { url: opts } : opts
      let sender = this.get
      if (method === 'POST') {
        sender = this.post
      }
      return new Promise((resolve, reject) => {
        sender.call(this, opts, (err, resp, body) => {
          if (err) reject(err)
          else resolve(resp)
        })
      })
    }

    get(opts) {
      return this.send.call(this.env, opts)
    }

    post(opts) {
      return this.send.call(this.env, opts, 'POST')
    }
  }

  return new (class {
    constructor(name, opts) {
      this.name = name
      this.http = new Http(this)
      this.data = null
      this.dataFile = 'box.dat'
      this.logs = []
      this.isMute = false
      this.isNeedRewrite = false
      this.logSeparator = '\n'
      this.encoding = 'utf-8'
      this.startTime = new Date().getTime()
      Object.assign(this, opts)
      this.log('', `🔔${this.name}, 开始!`)
    }

    isNode() {
      return 'undefined' !== typeof module && !!module.exports
    }

    isQuanX() {
      return 'undefined' !== typeof $task
    }

    isSurge() {
      return 'undefined' !== typeof $httpClient && 'undefined' === typeof $loon
    }

    isLoon() {
      return 'undefined' !== typeof $loon
    }

    toObj(str, defaultValue = null) {
      try {
        return JSON.parse(str)
      } catch {
        return defaultValue
      }
    }

    toStr(obj, defaultValue = null) {
      try {
        return JSON.stringify(obj)
      } catch {
        return defaultValue
      }
    }

    getjson(key, defaultValue) {
      let json = defaultValue
      const val = this.getdata(key)
      if (val) {
        try {
          json = JSON.parse(this.getdata(key))
        } catch {}
      }
      return json
    }

    setjson(value, key) {
      try {
        return this.setdata(JSON.stringify(value), key)
      } catch {
        return false
      }
    }

    getScript(url) {
      return new Promise((resolve) => {
        this.get({ url }, (err, resp, body) => resolve(body))
      })
    }

    runScript(script, runOpts) {
      return new Promise((resolve) => {
        let httpapi = this.getdata('@chavy_boxjs_userCfgs.httpapi')
        httpapi = httpapi ? httpapi.replace(/\n/g, '').trim() : httpapi
        let httpapi_timeout = this.getdata(
          '@chavy_boxjs_userCfgs.httpapi_timeout'
        )
        httpapi_timeout = httpapi_timeout ? httpapi_timeout * 1 : 20
        httpapi_timeout = runOpts && runOpts.timeout ? runOpts.timeout : httpapi_timeout
        const [key, addr] = httpapi.split('@')
        const opts = {
          url: `http://${addr}/v1/scripting/evaluate`,
          body: {
            script_text: script,
            mock_type: 'cron',
            timeout: httpapi_timeout
          },
          headers: { 'X-Key': key, 'Accept': '*/*' }
        }
        this.post(opts, (err, resp, body) => resolve(body))
      }).catch((e) => this.logErr(e))
    }

    loaddata() {
      if (this.isNode()) {
        this.fs = this.fs ? this.fs : require('fs')
        this.path = this.path ? this.path : require('path')
        const curDirDataFilePath = this.path.resolve(this.dataFile)
        const rootDirDataFilePath = this.path.resolve(
          process.cwd(),
          this.dataFile
        )
        const isCurDirDataFile = this.fs.existsSync(curDirDataFilePath)
        const isRootDirDataFile = !isCurDirDataFile && this.fs.existsSync(rootDirDataFilePath)
        if (isCurDirDataFile || isRootDirDataFile) {
          const datPath = isCurDirDataFile ? curDirDataFilePath : rootDirDataFilePath
          try {
            return JSON.parse(this.fs.readFileSync(datPath))
          } catch (e) {
            return {}
          }
        } else return {}
      } else return {}
    }

    writedata() {
      if (this.isNode()) {
        this.fs = this.fs ? this.fs : require('fs')
        this.path = this.path ? this.path : require('path')
        const curDirDataFilePath = this.path.resolve(this.dataFile)
        const rootDirDataFilePath = this.path.resolve(
          process.cwd(),
          this.dataFile
        )
        const jsondata = JSON.stringify(this.data)
        this.fs.writeFileSync(curDirDataFilePath, jsondata)

      }
    }
    lodash_get(source, path, defaultValue = undefined) {
      const paths = path.replace(/\[(\d+)\]/g, '.$1').split('.')
      let result = source
      for (const p of paths) {
        result = Object(result)[p]
        if (result === undefined) {
          return defaultValue
        }
      }
      return result
    }

    lodash_set(obj, path, value) {
      if (Object(obj) !== obj) return obj
      if (!Array.isArray(path)) path = path.toString().match(/[^.[\]]+/g) || []
      path
        .slice(0, -1)
        .reduce(
          (a, c, i) =>
            Object(a[c]) === a[c]
              ? a[c]
              : (a[c] = Math.abs(path[i + 1]) >> 0 === +path[i + 1] ? [] : {}),
          obj
        )[path[path.length - 1]] = value
      return obj
    }

    getdata(key) {
      let val = this.getval(key)
      if (/^@/.test(key)) {
        const [, objkey, paths] = /^@(.*?)\.(.*?)$/.exec(key)
        const obj = objkey ? this.getval(objkey) : ''
        if (obj) {
          try {
            const arr = JSON.parse(obj)
            val = arr ? this.lodash_get(arr, paths, '') : val
          } catch (e) {
            val = ''
          }
        }
      }
      return val
    }

    setdata(val, key) {
      let success = false
      if (/^@/.test(key)) {
        const [, objkey, paths] = /^@(.*?)\.(.*?)$/.exec(key)
        const obj = this.getval(objkey)
        const arr = obj ? JSON.parse(obj) : []
        this.lodash_set(arr, paths, val)
        success = this.setval(JSON.stringify(arr), objkey)
      } else {
        success = this.setval(val, key)
      }
      return success
    }

    getval(key) {
      if (this.isSurge() || this.isLoon()) {
        return $persistentStore.read(key)
      } else if (this.isQuanX()) {
        return $prefs.valueForKey(key)
      } else if (this.isNode()) {
        this.data = this.loaddata()
        return this.data[key]
      } else {
        return (this.data && this.data[key]) || null
      }
    }

    setval(val, key) {
      if (this.isSurge() || this.isLoon()) {
        return $persistentStore.write(val, key)
      } else if (this.isQuanX()) {
        return $prefs.setValueForKey(val, key)
      } else if (this.isNode()) {
        this.data = this.loaddata()
        this.data[key] = val
        this.writedata()
        return true
      } else {
        return (this.data && (this.data[key] = val)) || false
      }
    }

    initGotEnv(opts) {
      this.got = this.got ? this.got : require('got')
      this.cktough = this.cktough ? this.cktough : require('tough-cookie')
      this.ckjar = this.ckjar ? this.ckjar : new this.cktough.CookieJar()
      if (opts) {
        opts.headers = opts.headers ? opts.headers : {}
        if (undefined === opts.headers.Cookie && undefined === opts.cookieJar) {
          opts.cookieJar = this.ckjar
        }
      }
    }

    get(opts, callback = () => {}) {
      if (opts.headers) {
        delete opts.headers['Content-Type']
        delete opts.headers['Content-Length']
      }
      if (this.isSurge() || this.isLoon()) {
        if (this.isSurge() && this.isNeedRewrite) {
          opts.headers = opts.headers || {}
          Object.assign(opts.headers, { 'X-Surge-Skip-Scripting': false })
        }
        $httpClient.get(opts, (err, resp, body) => {
          if (!err && resp) {
            resp.body = body
            resp.statusCode = resp.status
          }
          callback(err, resp, body)
        })
      } else if (this.isQuanX()) {
        if (this.isNeedRewrite) {
          opts.opts = opts.opts || {}
          Object.assign(opts.opts, { hints: false })
        }
        $task.fetch(opts).then(
          (resp) => {
            const { statusCode: status, statusCode, headers, body } = resp
            callback(null, { status, statusCode, headers, body }, body)
          },
          (err) => callback(err)
        )
      } else if (this.isNode()) {
        this.initGotEnv(opts)
        this.got(opts)
          .on('redirect', (resp, nextOpts) => {
            try {
              if (resp.headers['set-cookie']) {
                const ck = resp.headers['set-cookie']
                  .map(this.cktough.Cookie.parse)
                  .toString()
                if (ck) {
                  this.ckjar.setCookieSync(ck, null)
                }
                nextOpts.cookieJar = this.ckjar
              }
            } catch (e) {
              this.logErr(e)
            }
          })
          .then(
            (resp) => {
              const { statusCode, headers, body } = resp
              callback(null, { statusCode, headers, body }, body)
            },
            (err) => {
              const { message: error, response: resp } = err
              callback(error, resp, resp && resp.body)
            }
          )
      }
    }

    post(opts, callback = () => {}) {
      const method = opts.method ? opts.method.toLocaleLowerCase() : 'post'
      if (opts.body && opts.headers && !opts.headers['Content-Type']) {
        opts.headers['Content-Type'] = 'application/x-www-form-urlencoded'
      }
      if (opts.headers) delete opts.headers['Content-Length']
      if (this.isSurge() || this.isLoon()) {
        if (this.isSurge() && this.isNeedRewrite) {
          opts.headers = opts.headers || {}
          Object.assign(opts.headers, { 'X-Surge-Skip-Scripting': false })
        }
        $httpClient[method](opts, (err, resp, body) => {
          if (!err && resp) {
            resp.body = body
            resp.statusCode = resp.status
          }
          callback(err, resp, body)
        })
      } else if (this.isQuanX()) {
        opts.method = method
        if (this.isNeedRewrite) {
          opts.opts = opts.opts || {}
          Object.assign(opts.opts, { hints: false })
        }
        $task.fetch(opts).then(
          (resp) => {
            const { statusCode: status, statusCode, headers, body } = resp
            callback(null, { status, statusCode, headers, body }, body)
          },
          (err) => callback(err)
        )
      } else if (this.isNode()) {
        this.initGotEnv(opts)
        const { url, ..._opts } = opts
        this.got[method](url, _opts).then(
          (resp) => {
            const { statusCode, headers, body } = resp
            callback(null, { statusCode, headers, body }, body)
          },
          (err) => {
            const { message: error, response: resp } = err
            callback(error, resp, resp && resp.body)
          }
        )
      }
    }

    time(format, ts = null) {
      const date = ts ? new Date(ts) : new Date()
      let o = {
        'M+': date.getMonth() + 1,
        'd+': date.getDate(),
        'H+': date.getHours(),
        'm+': date.getMinutes(),
        's+': date.getSeconds(),
        'q+': Math.floor((date.getMonth() + 3) / 3),
        'S': date.getMilliseconds()
      }
      if (/(y+)/.test(format))
        format = format.replace(
          RegExp.$1,
          (date.getFullYear() + '').substr(4 - RegExp.$1.length)
        )
      for (let k in o)
        if (new RegExp('(' + k + ')').test(format))
          format = format.replace(
            RegExp.$1,
            RegExp.$1.length == 1
              ? o[k]
              : ('00' + o[k]).substr(('' + o[k]).length)
          )
      return format
    }

    msg(title = name, subt = '', desc = '', opts) {
      const toEnvOpts = (rawopts) => {
        if (!rawopts) return rawopts
        if (typeof rawopts === 'string') {
          if (this.isLoon()) return rawopts
          else if (this.isQuanX()) return { 'open-url': rawopts }
          else if (this.isSurge()) return { url: rawopts }
          else return undefined
        } else if (typeof rawopts === 'object') {
          if (this.isLoon()) {
            let openUrl = rawopts.openUrl || rawopts.url || rawopts['open-url']
            let mediaUrl = rawopts.mediaUrl || rawopts['media-url']
            return { openUrl, mediaUrl }
          } else if (this.isQuanX()) {
            let openUrl = rawopts['open-url'] || rawopts.url || rawopts.openUrl
            let mediaUrl = rawopts['media-url'] || rawopts.mediaUrl
            let updatePasteboard =
              rawopts['update-pasteboard'] || rawopts.updatePasteboard
            return {
              'open-url': openUrl,
              'media-url': mediaUrl,
              'update-pasteboard': updatePasteboard
            }
          } else if (this.isSurge()) {
            let openUrl = rawopts.url || rawopts.openUrl || rawopts['open-url']
            return { url: openUrl }
          }
        } else {
          return undefined
        }
      }
      if (!this.isMute) {
        if (this.isSurge() || this.isLoon()) {
          $notification.post(title, subt, desc, toEnvOpts(opts))
        } else if (this.isQuanX()) {
          $notify(title, subt, desc, toEnvOpts(opts))
        }
      }
      if (!this.isMuteLog) {
        let logs = ['', '==============📣系统通知📣==============']
        logs.push(title)
        subt ? logs.push(subt) : ''
        desc ? logs.push(desc) : ''
        console.log(logs.join('\n'))
        this.logs = this.logs.concat(logs)
      }
    }

    log(...logs) {
      if (logs.length > 0) {
        this.logs = [...this.logs, ...logs]
      }
      console.log(logs.join(this.logSeparator))
    }

    logErr(err, msg) {
      const isPrintErr = !this.isMuteErr
      if (isPrintErr) {
        const error = err.stack ? err.stack : err
        console.log(`❗️${this.name}, 错误!`, error)
      }
    }

    wait(time) {
      return new Promise((resolve) => setTimeout(resolve, time))
    }

    done(val = {}) {
      const endTime = new Date().getTime()
      const costTime = (endTime - this.startTime) / 1000
      this.log('', `🔔${this.name}, 结束! 🕛 ${costTime} 秒`)
      this.log()
      if (this.isSurge() || this.isLoon()) {
        $done(val)
      } else if (this.isQuanX()) {
        $done(val)
      }
    }
  })(name, opts)
}

!(async () => {
    await getNotice();
    await main();
})().catch((e) => { $.log(e) }).finally(() => { $.done({}); });

async function main() {
    if (!EMSYHZX_OPENID) {
        $.log('请设置环境变量 EMSYHZX_OPENID');
        return;
    }
    
    let allUserNotice = '';
    const user_list = EMSYHZX_OPENID.split('&');
    
    for (const [index, openId] of user_list.entries()) {
        try {
            if (!openId) continue;
            
            notice = `\n--- 账号 ${index + 1} (${openId.slice(0, 10)}...) ---\n`;
            console.log(`\n--- 正在执行账号 ${index + 1} ---`);

            // 重置会话变量
            token = '';
            userId = '';
            
            const loginSuccess = await userLogin(openId);
            if (!loginSuccess) {
                allUserNotice += notice;
                await $.wait(2000); // 随机等待
                continue;
            }

            await $.wait(1000);
            await userSign(openId);
            
            await $.wait(1000);
            await querySignReward(openId);
            
            await $.wait(1000);
            await queryAccountInfo();

            allUserNotice += notice;
            
        } catch (e) {
            $.log(`账号执行出错: ${e}`);
        }
    }
    if (allUserNotice) {
        await sendMsg(allUserNotice);
    }
}

async function userLogin(openId) {
    console.log("尝试登录...");
    const body = {
        "appId": "wx52872495fb375c4b",
        "openId": openId,
        "source": "JD"
    };
    const result = await commonPost('/memberCenterApiV2/member/findByOpenIdAppId', body);
    if (result && result.code === '000000') {
        token = result.info.token;
        userId = result.info.memberId;
        console.log("🎉 登录成功!");
        notice += "登录状态: 成功 ✅\n";
        return true;
    } else {
        const errorMsg = result ? result.msg : "请求失败";
        console.log(`⛔ 登录失败: ${errorMsg}`);
        notice += `登录状态: 失败, ${errorMsg} ❌\n`;
        return false;
    }
}

async function userSign(openId) {
    console.log("执行签到...");
    const body = {
        "userId": userId,
        "appId": "wx52872495fb375c4b",
        "openId": openId,
        "activId": "c0c4a0a3ef8145f49f2e294741a3cd62"
    };
    const result = await commonPost('/activCenterApi/signActivInfo/sign', body);
    if (result && result.code === '000000') {
        console.log(`🎉 签到成功: 获得 ${result.info[0].prizeSize} 积分`);
        notice += `每日签到: 成功, 获得 ${result.info[0].prizeSize} 积分\n`;
    } else {
        const errorMsg = result ? result.msg : "请求失败";
        console.log(`⛔ 签到失败: ${errorMsg}`);
        notice += `每日签到: ${errorMsg}\n`;
    }
}

async function querySignReward(openId) {
    console.log("查询累计签到天数...");
    const body = {
        "userId": userId,
        "appId": "wx52872495fb375c4b",
        "openId": openId,
        "activId": "d191dce0740849b1b7377e83c00475d6"
    };
    const result = await commonPost('/activCenterApi/signActivInfo/querySignDetail', body);
    if (result && result.code === '000000') {
        console.log(`🎉 当前累计签到 ${result.info.signDay} 天`);
        notice += `累计签到: ${result.info.signDay} 天\n`;
    } else {
        const errorMsg = result ? result.msg : "请求失败";
        console.log(`⛔ 查询累计签到失败: ${errorMsg}`);
        notice += `累计签到: 查询失败\n`;
    }
}

async function queryAccountInfo() {
    console.log("查询账户总积分...");
    const result = await commonPost('/memberCenterApiV2/golds/memberGoldsInfo', {});
    if (result && result.code === '000000') {
        console.log(`🎉 查询成功: 总积分 ${result.info.availableGoldsTotal}`);
        notice += `当前积分: ${result.info.availableGoldsTotal}\n`;
    } else {
        const errorMsg = result ? result.msg : "请求失败";
        console.log(`⛔ 查询积分失败: ${errorMsg}`);
        notice += `当前积分: 查询失败\n`;
    }
}

async function commonPost(url, body = {}) {
    let attempt = 0;
    while (attempt < maxRetries) {
        attempt++;
        const result = await new Promise(resolve => {
            const options = {
                url: `https://ump.ems.com.cn${url}`,
                headers: {
                    "Content-Type": 'application/json',
                    'MC-TOKEN': token, // 使用全局变量token
                    "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 17_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 MicroMessenger/8.0.48(0x18003030) NetType/WIFI Language/zh_CN",
                },
                body: JSON.stringify(body)
            };
            $.post(options, async (err, resp, data) => {
                try {
                    if (err) {
                        console.log(`请求错误: ${JSON.stringify(err)}`);
                        console.log(`${$.name} API请求失败，请检查网络`);
                    } else {
                        resolve(JSON.parse(data));
                    }
                } catch (e) {
                    $.logErr(e, resp);
                } finally {
                    resolve(null); // 确保在任何情况下都resolve
                }
            });
        });
        
        if (result) {
            return result;
        } else if (attempt < maxRetries) {
            console.log(`请求失败，正在进行第 ${attempt} 次重试...`);
            await $.wait(2000); // 等待2秒后重试
        }
    }
    return null; // 所有重试都失败后返回 null
}


async function getNotice() {
    try {
        const options = { url: `https://fastly.jsdelivr.net/gh/xzxxn777/Surge@main/Utils/Notice.json` };
        const resp = await $.get(options);
        if (resp.body) {
            console.log(JSON.parse(resp.body).notice);
        }
    } catch (e) {
        // 静默处理，获取公告失败不影响主流程
    }
}

async function sendMsg(message) {
    if (!message) return;
    if ($.isNode()) {
        await notify.sendNotify($.name, message);
    } else {
        $.msg($.name, '', message);
    }
}

File: EMS邮惠中心.py
"""
作者: Mist (由临渊模板改造)
日期: 2025/08/13
name: code版_EMS邮惠中心
入口: EMS邮政快递小程序
功能: 签到、查询连续签到、查询积分
变量: emsyhzx_data (openId) 多个账号用换行或#分割
定时: 一天一次
cron: 0 9 * * *
------------更新日志------------
2025/8/13   V1.0    基于临渊模板V1.1进行适配改造
"""

import json
import random
import time
import requests
import os
import traceback
import ssl

MULTI_ACCOUNT_SPLIT = ["\n", "#"] # 分隔符列表
MULTI_ACCOUNT_PROXY = False # 是否使用多账号代理，默认不使用，True则使用多账号代理
NOTIFY = os.getenv("LY_NOTIFY") or False # 是否推送日志，默认不推送，True则推送

class AutoTask:
    def __init__(self, script_name):
        """
        初始化自动任务类
        :param script_name: 脚本名称，用于日志显示
        """
        self.script_name = script_name
        self.log_msgs = []
        self.proxy_url = os.getenv("PROXY_API_URL") # 代理api，返回一条txt文本，内容为代理ip:端口
        self.wx_appid = "wx52872495fb375c4b" # EMS小程序id
        self.host = "ump.ems.com.cn"
        self.token = ""
        self.user_id = ""
        self.nickname = ""
        self.user_agent = "Mozilla/5.0 (iPhone; CPU iPhone OS 17_4_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 MicroMessenger/8.0.48(0x18003030) NetType/WIFI Language/zh_CN"
        
    def log(self, msg, level="info"):
        """
        记录并打印日志
        """
        # 获取当前时间
        current_time = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
        # 格式化日志消息
        formatted_msg = f"[{current_time}] [{level.upper()}] {msg}"
        print(formatted_msg)
        self.log_msgs.append(msg) # 保存原始消息用于通知

    def get_proxy(self):
        """
        获取代理
        :return: 代理
        """
        if not self.proxy_url:
            self.log("[获取代理] 没有找到环境变量PROXY_API_URL，不使用代理", level="warning")
            return None
        try:
            url = self.proxy_url
            response = requests.get(url, timeout=5)
            response.raise_for_status()
            proxy = response.text
            self.log(f"[获取代理] {proxy}")
            return proxy
        except Exception as e:
            self.log(f"[获取代理] 获取代理失败: {e}", level="error")
            return None

    def check_env(self):
        """
        检查环境变量
        :return: 环境变量迭代器
        """
        try:
            # 从环境变量获取
            env_data = os.getenv("emsyhzx_data")
            if not env_data:
                self.log("[检查环境变量] 没有找到环境变量emsyhzx_data，请检查环境变量", level="error")
                return None

            # 自动检测分隔符
            split_char = None
            for sep in MULTI_ACCOUNT_SPLIT:
                if sep in env_data:
                    split_char = sep
                    break
            
            accounts = [env_data] if not split_char else env_data.split(split_char)

            for account in accounts:
                if account.strip():
                    yield account.strip()
        except Exception as e:
            self.log(f"[检查环境变量] 发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
            raise
        
    def login(self, session, open_id):
        """
        使用openId登录
        :param session: session
        :param open_id: 用户的openId
        :return: 登录结果
        """
        try:
            url = f"https://{self.host}/memberCenterApiV2/member/findByOpenIdAppId"
            payload = {
                "appId": self.wx_appid,
                "openId": open_id,
                "source": "JD"
            }
            response = session.post(url, json=payload, timeout=15)
            response_json = response.json()
            if response_json.get('code') == "000000":
                self.token = response_json['info']['token']
                self.user_id = response_json['info']['memberId']
                self.nickname = f"用户_{str(self.user_id)[-4:]}" # 构造一个昵称
                session.headers['MC-TOKEN'] = self.token
                self.log(f"[{self.nickname}] 登录成功")
                return True
            else:
                self.log(f"[登录] 失败，错误信息: {response_json.get('msg', '未知错误')}", level="error")
                return False
        except requests.RequestException as e:
            self.log(f"[登录] 发生网络错误: {str(e)}\n{traceback.format_exc()}", level="error")
            return False
        except Exception as e:
            self.log(f"[登录] 发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
            return False
        
    def sign_in(self, session, open_id):
        """
        签到
        :param session: session
        :param open_id: 用户的openId
        """
        try:
            url = f"https://{self.host}/activCenterApi/signActivInfo/sign"
            payload = {
                "userId": self.user_id,
                "appId": self.wx_appid,
                "openId": open_id,
                "activId": "c0c4a0a3ef8145f49f2e294741a3cd62"
            }
            response = session.post(url, json=payload, timeout=15)
            response_json = response.json()
            if response_json.get('code') == "000000":
                prize_size = response_json.get('info', [{}])[0].get('prizeSize', '未知')
                self.log(f"[{self.nickname}] 签到成功，获得[{prize_size}]积分")
                return True
            else:
                self.log(f"[{self.nickname}] 签到失败: {response_json.get('msg', '未知错误')}", level="warning")
                return False
        except Exception as e:
            self.log(f"[{self.nickname}] 签到时发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
            return False
        
    def check_cumulative_sign_in(self, session, open_id):
        """
        查询连续签到天数
        :param session: session
        :param open_id: 用户的openId
        """
        try:
            url = f"https://{self.host}/activCenterApi/signActivInfo/querySignDetail"
            payload = {
                "userId": self.user_id,
                "appId": self.wx_appid,
                "openId": open_id,
                "activId": "d191dce0740849b1b7377e83c00475d6",
            }
            response = session.post(url, json=payload, timeout=15)
            response_json = response.json()
            if response_json.get('code') == '000000':
                sign_day = response_json.get('info', {}).get('signDay', '未知')
                self.log(f"[{self.nickname}] 当前累计签到: [{sign_day}]天")
                return True
            else:
                self.log(f"[{self.nickname}] 查询累计签到失败: {response_json.get('msg', '未知错误')}", level="error")
                return False
        except Exception as e:
            self.log(f"[{self.nickname}] 查询累计签到时发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
            return False
        
    def get_user_points(self, session):
        """
        获取用户积分
        :param session: session
        """
        try:
            url = f"https://{self.host}/memberCenterApiV2/golds/memberGoldsInfo"
            response = session.post(url, json={}, timeout=15)
            response_json = response.json()
            if response_json.get('code') == '000000':
                points = response_json.get('info', {}).get('availableGoldsTotal', '未知')
                self.log(f"[{self.nickname}] 查询成功，当前总积分: [{points}]")
                return True
            else:
                self.log(f"[{self.nickname}] 查询积分失败: {response_json.get('msg', '未知错误')}", level="error")
                return False
        except Exception as e:
            self.log(f"[{self.nickname}] 查询积分时发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
            return False
        
    def run(self):
        """
        运行任务
        """
        try:
            self.log(f"【{self.script_name}】开始执行任务")
            accounts = self.check_env()
            if not accounts:
                return

            for index, open_id in enumerate(accounts, 1):
                # 清理账号信息
                self.nickname = ""
                self.token = ""
                self.user_id = ""
                self.log("")
                self.log(f"------ 【账号{index}】开始执行任务 ------")
                
                session = requests.Session()
                headers = {
                    "User-Agent": self.user_agent,
                    "Content-Type": 'application/json',
                    "authority": self.host
                }
                session.headers.update(headers)

                if MULTI_ACCOUNT_PROXY:
                    proxy = self.get_proxy()
                    if proxy:
                        session.proxies.update({"http": f"http://{proxy}", "https": f"http://{proxy}"})

                # 1. 登录
                if not self.login(session, open_id):
                    self.log(f"账号{index} 登录失败，跳过该账号", level="error")
                    session.close()
                    continue
                
                time.sleep(random.randint(2, 4))
                
                # 2. 签到
                self.sign_in(session, open_id)
                time.sleep(random.randint(2, 4))

                # 3. 查询连续签到
                self.check_cumulative_sign_in(session, open_id)
                time.sleep(random.randint(2, 4))
                
                # 4. 查询积分
                self.get_user_points(session)
                
                self.log(f"------ 【账号{index}】执行任务完成 ------")
                # 清理session
                session.close()

        except Exception as e:
            self.log(f"【{self.script_name}】执行过程中发生错误: {str(e)}\n{traceback.format_exc()}", level="error")
        finally:
            if NOTIFY:
                # 如果notify模块不存在，从远程下载至本地
                if not os.path.exists("notify.py"):
                    try:
                        url = "https://raw.githubusercontent.com/whyour/qinglong/refs/heads/develop/sample/notify.py"
                        response = requests.get(url)
                        response.raise_for_status()
                        with open("notify.py", "w", encoding="utf-8") as f:
                            f.write(response.text)
                        self.log("下载notify.py成功")
                        import notify
                    except Exception as e:
                        self.log(f"下载notify.py失败: {e}", level="error")
                else:
                    import notify
                
                # 任务结束后推送日志
                try:
                    title = f"{self.script_name} 运行日志"
                    header = "作者：Mist (由临渊模板改造)\n"
                    content = header + "\n" +"\n".join(self.log_msgs)
                    notify.send(title, content)
                    self.log("日志已通过notify发送")
                except NameError:
                    self.log("notify模块未成功导入，无法发送通知", level="warning")
                except Exception as e:
                    self.log(f"通过notify发送日志时发生错误: {e}", level="error")


if __name__ == "__main__":
    auto_task = AutoTask("EMS邮惠中心")
    auto_task.run()